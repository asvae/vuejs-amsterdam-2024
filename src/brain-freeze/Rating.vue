<template>
  <div class="flex" role="radiogroup">
    <div
        v-for="index in 5"
        :key="index"
        class="cursor-pointer py-2"
        :class="{'text-yellow-400': index <= model, 'text-gray-400': index > model}"
        @click="model = index"
        role="radio"
        :aria-checked="index <= model.value ? 'true' : 'false'"
        :aria-label="'Rating ' + index"
    >
      â˜…
    </div>
  </div>
</template>


<script setup>
const model = defineModel({default: 1})
</script>
