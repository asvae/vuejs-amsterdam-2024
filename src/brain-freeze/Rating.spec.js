import { describe, it, expect } from 'vitest';
import { mount } from '@vue/test-utils';
import Rating from './Rating.vue';

describe("Rating", () => {
  it("updates modelValue when a rating item is clicked", async () => {
    const wrapper = mount(Rating, {
      props: {
        modelValue: 2,
      },
      attrs: {
        "onUpdate:modelValue": (newValue) => wrapper.setProps({ modelValue: newValue }),
      },
    });

    const newRating = 3; // The new rating to be selected
    await wrapper.findAll('[role="radio"]')[newRating - 1].trigger('click');
    expect(wrapper.props("modelValue")).toBe(newRating);
  });

  it("matches the snapshot", () => {
    const wrapper = mount(Rating, {
      props: {
        modelValue: 3,
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
