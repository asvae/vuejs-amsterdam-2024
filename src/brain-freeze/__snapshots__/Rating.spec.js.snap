// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Rating > matches the snapshot 1`] = `
"<div role="radiogroup">
  <div class="cursor-pointer py-2 text-yellow-400" role="radio" aria-checked="false" aria-label="Rating 1"> ★ </div>
  <div class="cursor-pointer py-2 text-yellow-400" role="radio" aria-checked="false" aria-label="Rating 2"> ★ </div>
  <div class="cursor-pointer py-2 text-gray-400" role="radio" aria-checked="false" aria-label="Rating 3"> ★ </div>
  <div class="cursor-pointer py-2 text-gray-400" role="radio" aria-checked="false" aria-label="Rating 4"> ★ </div>
  <div class="cursor-pointer py-2 text-gray-400" role="radio" aria-checked="false" aria-label="Rating 5"> ★ </div>
</div>"
`;
